### ISSUE TITLE
Does not work with `new` objects

### ISSUE THREAD

**Issue by NightJar** [silverstripe/silverstripe-upgrader#181]
Running `upgrade-code ss3module modulename/src` does not convert e.g. `new CompositeField` to `CompositeField::create()`... but probably more to the point, it does not add `use SilverStripe\Forms\CompositeField;` atop the class that is being upgraded.

**Comment by tractorcow** [comment]
Good idea; we could use injector detection (`in_array(Injectable::class, class_uses($class))`) to automatically convert these.

However, you need to be careful that you don't assume this is correct. E.g. code inside an Injector factory should be omitted (as it is already within an injection context).

**Comment by NightJar** [comment]
The more critical element, which I've failed to emphasise correctly, is that the upgrader fails to upgrade an entire file if it already has a namespace.

The class in question in this case started out `class Thing extends DataExtension`

After `upgrade-code upgrade` it went something like so:

```php
<?

namespace TheVendor\TheModule\Thing;



use DateTime;


class Thing extends DataExtension
{
```

**Comment by robbieaverill** [comment]
There should be another command that actually implements the namespaces, did you run the full upgrade process?

**Comment by maxime-rainville** [comment]
The `upgrade` command should have added a `use` statement for `CompositeField`. Do other class get pick up as expected?

**Comment by NightJar** [comment]
Ah, this is the trick.
I ran add namespace recursive. Since I had already added the namespace it skipped it.
It seems then perhaps there's some conflation in the goal of this step - that is to say it does more than one thing.

It add a `namespace` directive. This is as the name implies.
It parses all code and adds `use` statements for existing class type tokens.

If the first item does not complete, then the second is skipped.
I feel this to be incorrect behaviour.

**Comment by maxime-rainville** [comment]
I think the convert `new` to `create` and the missing `use` statement are two different problem. Do you want to raise a separate issue for it?
